marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -367,14 +366,6 @@ endToEnd = H.withShrinks 0 $ H.integration $ (liftIO TN.setDarwinTmpdir >>) $ HE
       keyWitnesses' = map (C.makeShelleyKeyWitness txBody) keyWitnesses
   TN.submitTx localNodeConnectInfo $ C.makeSignedTransaction keyWitnesses' txBody
 
-  -- Receive event from the indexer, compare the mint that we
```

Comment:
Why bin this?

Reply:
The callback type was wrong: it expected an `Event`, while we need to pass it an indexer to be able to release the indexer (if I understand correctly how the callback worked).

Reply:
Since you removed it, where is the assertion now?

Also, I encountered this in the past, and the solution would be to have as parameter to the callback (indexer, event).

Reply:
Then I'd be in favour of removing the test entirely. Passing along the event for testing purpose sounds very bad, unless I miss the point.

Reply:
But receiving the most recent event via `onInsert` isn't bad in itself, perhaps it's just not codified into a type class. I.e maybe all indexers could pass it the event that was inserted and the indexer. (How does the new implementation handle this?)

We would like to continue to test that an indexer indexes the events that it's supposed to index, so removing the test doesn't sound all that good either..

Reply:
The test itself doesn't check that the event is stored, it checks that we pass it to the callback. It's useful if we want to pass the last event in the callback, for which we have no use case as the callback is mainly to pass an updated indexer to the query side.
If we pass the resulting indexer in the callback, we can check if its content has the forged mint. I can update the test in this direction.

(For the new indexer, on a worker creation, we create an MVar that is output in the result of the creation function, and that is updated by the worker, so there's no callback anymore. Test for insertion can be done through queries)

Reply:
:+1:, yup querying the indexer would be good.

Reply:
@berewt I see you're on PTO -- I can take a crack at this particular issue if @koslambrou doesn't mind.

I think that querying the event instead of passing the event to `onInsert` is an even better way to do it (and hopefully shouldn't be very difficult).

Reply:
I am, and I'm fine with this plan (you meant querying the _indexer_, right?)

Reply:
> I see you're on PTO -- I can take a crack at this particular issue if @koslambrou doesn't mind.

@eyeinsky Yep that would be good